util/http.py

```diff
@@ -27,7 +27,8 @@ def __init__(self, url_or_service, message, debug_message=None):
            (e.g. "Overdrive"), or the specific URL that had the problem.
         """
         super(RemoteIntegrationException, self).__init__(message)
-        if any(url_or_service.startswith(x) for x in ('http:', 'https:')):
+        if (url_or_service and
```

Comment:
Sometimes `url_or_service` isn't set. Usually during testing, but it could happen for real (if we try to connect to a server but the hostname is specified as None). There's no reason why that should cause an error here.